packages/mobile/src/screens/collection-screen/useCollectionLineup.ts

```diff
@@ -0,0 +1,114 @@
+import { useCallback } from 'react'
+
+import type { SmartCollectionVariant } from '@audius/common'
+import {
+  useProxySelector,
+  Kind,
+  makeUid,
+  cacheActions,
+  cacheCollectionsSelectors,
+  collectionPageLineupActions,
+  collectionPageSelectors,
+  lineupSelectors,
+  reachabilitySelectors
+} from '@audius/common'
+import moment from 'moment'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { useIsOfflineModeEnabled } from 'app/hooks/useIsOfflineModeEnabled'
+import { useReachabilityEffect } from 'app/hooks/useReachabilityEffect'
+import { store } from 'app/store'
+import { getOfflineTracks } from 'app/store/offline-downloads/selectors'
+
+const { getCollection } = cacheCollectionsSelectors
+const { getCollectionTracksLineup } = collectionPageSelectors
+const { makeGetTableMetadatas } = lineupSelectors
+const { getIsReachable } = reachabilitySelectors
+
+const getTracksLineup = makeGetTableMetadatas(getCollectionTracksLineup)
+
+/**
+ * Returns a collection lineup, supports boths online and offline
+ * @param collectionId the numeric collection id
+ */
+export const useCollectionLineup = (
+  collectionId: number | SmartCollectionVariant | null,
+  fetchLineup: () => void
+) => {
+  const dispatch = useDispatch()
+  const isOfflineModeEnabled = useIsOfflineModeEnabled()
+  const offlineTracks = useSelector(getOfflineTracks)
+  const isReachable = useSelector(getIsReachable)
+  const collection = useSelector((state) => {
+    return getCollection(state, { id: collectionId as number })
+  })
+  const collectionTracks = useSelector(getCollectionTracksLineup)
+  const collectionTrackUidMap = collectionTracks.entries.reduce(
```

Comment:
would it be more performant to make the trackUidMap in the useSelector? maybe need an equality check there or somethnig?

Reply:
Yeah probably, this isn't causing rerenders but probably expensive to compute on every render